server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/RulesIntegrationTest.java

```diff
@@ -0,0 +1,364 @@
+        AccessToken user2AccessToken = authenticateJamesUser(baseUri(guiceJamesServer), USER_2, PASSWORD);
+        // User 1 orpheus@mydomain.tld sets a rule set
+        setEffectiveRule(user1AccessToken);
+
+        // When
+        // User 2 eurydice@mydomain.tld sends User 1 a mail
+        sendNotMatchingMail(user2AccessToken, getOutboxId(user2AccessToken), "user|inbox|1");
+
+        // Then
+        // User 1 should well receive this mail in the filtered mailbox
+        calmlyAwait.atMost(30, TimeUnit.SECONDS)
+            .until(() -> isTextMessageReceived(user1AccessToken, getInboxId(user1AccessToken), ORIGINAL_MESSAGE_TEXT_BODY, USER_2, USER_1));
+    }
+
+    @Test
+    public void jmapRuleShouldNotFilterAMailWhenNotSet() throws Exception {
+        /* Test scenario :
+            - User 2 eurydice@mydomain.tld sends User 1 a mail
+            - User 1 should well receive this mail in the inbox mailbox
+        */
+
+        // Given
+        AccessToken user1AccessToken = authenticateJamesUser(baseUri(guiceJamesServer), USER_1, PASSWORD);
+        AccessToken user2AccessToken = authenticateJamesUser(baseUri(guiceJamesServer), USER_2, PASSWORD);
+
+        // When
+        // User 2 eurydice@mydomain.tld sends User 1 a mail
+        sendMatchingMail(user2AccessToken, getOutboxId(user2AccessToken), "user|inbox|1");
+
+        // Then
+        // User 1 should well receive this mail in the filtered mailbox
+        calmlyAwait.atMost(30, TimeUnit.SECONDS)
+            .until(() -> isTextMessageReceived(user1AccessToken, getFilteredMailboxId(user1AccessToken), ORIGINAL_MESSAGE_TEXT_BODY, USER_2, USER_1));
+    }
+
+    @Test
+    public void jmapRuleShouldNotFilterAMailWhenSetEmpty() throws Exception {
+        /* Test scenario :
+            - User 1 orpheus@mydomain.tld sets an empty rule set
+            - User 2 eurydice@mydomain.tld sends User 1 a mail
+            - User 1 should well receive this mail in the inbox mailbox
+        */
+
+        // Given
+        AccessToken user1AccessToken = authenticateJamesUser(baseUri(guiceJamesServer), USER_1, PASSWORD);
+        AccessToken user2AccessToken = authenticateJamesUser(baseUri(guiceJamesServer), USER_2, PASSWORD);
+        // User 1 orpheus@mydomain.tld sets an empty rule set
+        setEmptyRule(user1AccessToken);
+
+        // When
+        // User 2 eurydice@mydomain.tld sends User 1 a mail
+        sendMatchingMail(user2AccessToken, getOutboxId(user2AccessToken), "user|inbox|1");
+
+        // Then
+        // User 1 should well receive this mail in the filtered mailbox
+        calmlyAwait.atMost(30, TimeUnit.SECONDS)
+            .until(() -> isTextMessageReceived(user1AccessToken, getFilteredMailboxId(user1AccessToken), ORIGINAL_MESSAGE_TEXT_BODY, USER_2, USER_1));
+    }
+
+    @Test
+    public void jmapRuleShouldFilterAMatchingMailWhenSetEmptyThenEffectively() throws Exception {
+        /* Test scenario :
```

Comment:
This feels like tests for GITLAB-734

In a first time we should simply exercise the CRUD (get/set methods)